_posts/2015-10-19-introduction_to_clojurescript.md

```diff
@@ -0,0 +1,269 @@
+
+### Installing Leiningen
+Leiningen is a build tool for Clojure projects. It will 
+http://leiningen.org/#install
+
+`brew install leiningen`
+
+### Using the Repl
+Now that we have Leningen installed, we can start getting familiar with ClojureScript syntax.
+
+Execute the following command at a terminal: `lein repl`
+
+You should get output that looks similar to:
+
+```
+$ lein repl
+nREPL server started on port 58371 on host 127.0.0.1 - nrepl://127.0.0.1:58371
+REPL-y 0.3.7, nREPL 0.2.10
+Clojure 1.7.0
+Java HotSpot(TM) 64-Bit Server VM 1.6.0_65-b14-466.1-11M4716
+    Docs: (doc function-name-here)
+          (find-doc "part-of-name-here")
+  Source: (source function-name-here)
+ Javadoc: (javadoc java-object-or-class-here)
+    Exit: Control+D or (exit) or (quit)
+ Results: Stored in vars *1, *2, *3, an exception in *e
+
+user=> 
+```
+
+We're now in a ClojureScript Repl. This will allow us quickly execute ClojureScript and view the result.
+
+
+## ClojureScript Syntax
+ClojureScript is a functional language. This means it has functions and limited additional language constructs. It is a Lisp style language.
+
+### Primitives
+
+* Number
+
+    ```
+    user=> 1.23
+    1.23
+    ```
+
+
+* String
+
+    ```
+    user=> "foo"
+    "foo"
+    ```
+
+
+* Vector (array)
+    
+    ```
+    user=> [:bar 3.14 "hello"]
+    [:bar 3.14 "hello"]
+    ```
+
+
+* Map (associative arrays)
+
+    ```
+    user=> {:msg "hello" :pi 3.14 :primes [2 3 5 7 11 13]}
+    {:msg "hello", :pi 3.14, :primes [2 3 5 7 11 13]}
+    ```
+
+* Keyword (used to access Maps)
+
+    ```
+    user=> :foo
+    :foo
+    ```
+
+
+* Set (distinct array)
+    
+    ```
+    user=> #{:bar 3.14 "hello"}
+    #{"hello" 3.14 :bar}
+    ```
+
+### Functions all the way down
+Functions are the building blocks of ClojureScript. To define a function, you even call a function `defn`.
+
+```
+user=> (defn myfunction [argument1] argment1)
+```
+Here we've defined a function named `myfunction`. It takes one argument `argument1` and returns it. It's not a very useful function but it's a good example of syntax. This is equivalent to the following Javascript:
+
+```javascript
+function myfunction(argument1){
+    return argument1;
+}
+```
+
+Functions are invoked by wrapping their name and arguments with parentheses:
+
+```
+user=> (myfunction "hello world")
+"hello world"
+```
+
+In non-functional programming languages there are special "Operators" or keywords. 
+In Javascript, some commonly used operators are are `+ - == if`. In ClojureScript and other Lisp languages, there are no special operators. These are just regular functions. 
+
+If statements are functions
+
+```
+user=> (if true "do true stuff here" "do false stuff here")
+"do true stuff here"
+```
+
+Math operators are functions
+
+```
+user=> (+ 2 3)
+5
+user=> (* 2 3)
+6
+```
+
+For more great examples of [Javascript to ClojureScript synonyms checkout this site](https://kanaka.github.io/clojurescript/web/synonym.html)
```

Comment:
might be useful here to say how to exit the repl, since next command is not in repl. even tho i see it's in the startup message now I ignored it.